include/boost/url/impl/url.ipp

```diff
@@ -1801,17 +1786,15 @@ set_fragment(
         this->string());
     s = buf.maybe_copy(s);
     check_invariants();
-    static constexpr auto cs =
-        pchars + '/' + '?';
```

Comment:
Good, you got rid of this function-local static and used the variable that was already there?

Reply:
Yes. Everything is in `boost/url/rfc/detail/charsets.hpp` now. This also avoids accidents, because these charsets were being replicated at multiple places.